### PR TITLE
apply to multi-cag parameters

### PR DESCRIPTION
**Pull request by Yanzhiwen001**
Feature: Multi-cag parameter as input
Add label to indicate each value for multi catagories  (in yaml)
Improve command line generation with multi catagories (in scallop_casmopolitan_yaml.py)

### CODE REVIEW

File: BO_training/main.py [hunk 1]
```diff
@@ -21,6 +21,7 @@
 parser.add_argument('--batch_size', type=int, default=1, help='batch size for BO.')
 parser.add_argument('--n_trials', type=int, default=20, help='number of trials for the experiment')
 parser.add_argument('--n_init', type=int, default=20, help='number of initialising random points')
+parser.add_argument('--failtol', type=int, default=18, help='stop condition')
```
Comment (gmarcais) [inline BO_training/main.py]: These option names or help are a bit cryptic. I don't know what it means.

File: BO_training/mixed_test_func/scallop_casmopolitan_yaml.py [hunk 2]
```diff
@@ -174,9 +176,7 @@ def __init__(self, input_file, normalize=False, boundary_fold = 0,ref_file=''):
         self.lb = np.array(lb)
         self.ub = np.array(ub)
         
-        #TODO: now only support binary cag parameters
-        #original code: self.vertices = np.array([2,2])
-        self.n_vertices = np.array(len(categorical_dims)*[2])
+        #TODO: now support multi cag parameters
```
Comment (gmarcais) [inline BO_training/mixed_test_func/scallop_casmopolitan_yaml.py]: If the support is done, remove the "TODO"

File: scallop.yml [hunk 3]
```diff
@@ -2,44 +2,67 @@
...
+#formats the specified value(s) and insert them inside the string's placeholder.
+#The placeholder is defined using curly brackets: {}
+#The placeholder should appear excatly once in the same section
+testing_software:
+  format: '{path} {input_option} {input_file} {parameters} {output_option} {pid}{output_format} {additional_option}'
+  input_option: '-i'
+  output_option: '-o'
+  output_format: '.gtf'
+  pid: ''
+  additional_option: ' --verbose 0 --min_transcript_coverage 0 --library_type empty > /dev/null 2>&1'
```
Comment (gmarcais) [inline scallop.yml:15]: Sending to `/dev/null` can create silent errors. Is there a better way to save those logs? Or are they really useless?

Comment (Yanzhiwen001) [reply scallop.yml:15]: Not sure if there is any silent error I missed, I will run default parameters without /dev/null on scallop and scallop2 to see if there is any. 
But these logs are indeed useless (if no error within them) BO computes AUC based following results produced by gffcompare and gtfcuff, which only depends on GTF files not log files.

**Review by gmarcais — APPROVED** [review]

**Pull request merged**